### ISSUE TITLE
Append an operation name to each OpenTelemetry activity

### ISSUE THREAD

**Issue by bh3605** [Particular/NServiceBus#6720]
#### Is your feature related to a problem? Please describe.
Enabling Open Telemetry support creates telemetry in app insights with no operation name. They bucket in a single <Empty> operation.

![image](https://user-images.githubusercontent.com/3073436/233176587-da24777c-1747-47d2-a625-e130b698a786.png)

#### Describe the requested feature
        
Append an operation name to the activity. Could be "EndpointName: send message", "EndpointName: publish event", etc.

#### Describe alternatives you've considered
Rolling my own implementation

**Comment by ramonsmits** [comment]
@bh3605 What view is this in application insights?

As you can see in the following sample there is activity name information set in the activities:

- https://docs.particular.net/samples/open-telemetry/jaeger/

We have a set of fixes names we use:

- https://github.com/Particular/NServiceBus/blob/c84561af72649ef1313dd3083a03847a5fe72e2c/src/NServiceBus.Core/OpenTelemetry/Tracing/ActivityNames.cs#L5-L10

**Comment by bh3605** [comment]
It's the performance view
![image](https://user-images.githubusercontent.com/3073436/233693594-928235d0-630e-4d71-ab1d-8613b1193e38.png)

I do see the activity names in the end-to-end transaction details view
![image](https://user-images.githubusercontent.com/3073436/233693840-792deb42-1fab-441a-93fd-654f5d964730.png)

**Comment by ramonsmits** [comment]
Thanks @bh3605 for that information. I'm not familiar with that screen. It seems we are providing the operation name so a little research seems needed to understand how that data is presented in this application insights view and what additional properties/tags we need to set.

I'll mark this as an improvement for a future version.

**Comment by bh3605** [comment]
Thanks! You can see I was using your sample so reproducing this is straight forward as long as you have an application insights instance to send data too.

I believe what you're going to have to do is at a point a message is sent you need to call StartOperation("", ....) and thereafter you can associate Activity objects with the current operation context.

Helpful link
https://learn.microsoft.com/en-us/azure/azure-monitor/app/custom-operations-tracking#applicationinsights-operations-vs-systemdiagnosticsactivity
Your distributed tracing sample has the operation name assigned. If you combined that solution with what you have you ought to be gravy. https://docs.particular.net/samples/tracing/

**Comment by bh3605** [comment]
@ramonsmits were you able to find anything?

**Comment by ramonsmits** [comment]
@bh3605 I see that no progress was made on this. However, we did made changes related to OpenTelemetry. 

Are you still affected by this?

**Comment by bh3605** [comment]
I don't believe so. I'm been enjoying the telemetry correlation and each operation has its own name.
